### PR TITLE
[AzureResourceManagerTemplateDeployment] Updating azure-arm-rest-v2 package

### PR DESCRIPTION
**Pull request by NinadKavimandan**
**Task name**: AzureResourceManagerTemplateDeployment@3

**Description**: Not setting `endpoint.url` to `graphurl` when `useGraphActiveDirectoryResource` is true. And avoiding resetting of endpoint.activeDirectoryResourceID. This PR is for upgrading the azure-arm-rest-v2 package, will raise another PR for task version bump.

**Documentation changes required:** No

**Added unit tests:** No

**Attached related issue:** [AzureStack] ICM: Wrong Endpoint URL #13623 

**Checklist**:
- [ ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Review by damccorm — APPROVED** [review]
LGTM, though I can't speak to the typescript changes

File: common-npm-packages/azure-arm-rest-v2/azure-arm-endpoint.ts [hunk 1]
```diff
@@ -96,7 +95,7 @@ export class AzureRMEndpoint {
                 }
                 else {
                     this.endpoint.environmentAuthorityUrl = (!!this.endpoint.environmentAuthorityUrl) ? this.endpoint.environmentAuthorityUrl : "https://login.windows.net/";
-                    this.endpoint.activeDirectoryResourceID = this.endpoint.url;
+                    if (!useGraphActiveDirectoryResource) this.endpoint.activeDirectoryResourceID = this.endpoint.url;
```
Comment (bishal-pdMSFT) [inline common-npm-packages/azure-arm-rest-v2/azure-arm-endpoint.ts:98]: please use braces to keep it shorter lines and more readable.

**Comment by bishal-pdMSFT** [comment]
@nadesu @chshrikh can someone from your team also review this as it fixes an issue with Azure stack?

**Review by bishal-pdMSFT — APPROVED** [review]

File: common-npm-packages/azure-arm-rest-v2/azure-arm-endpoint.ts (hunk 1 above)
Comment (NinadKavimandan) [reply common-npm-packages/azure-arm-rest-v2/azure-arm-endpoint.ts]: added braces with next commit

**Comment by bishal-pdMSFT** [comment]
@chshrikh / @nadesu can somebody please review this PR as we need to hotfix this for an urgent DTS issue.

**Comment by nadesu** [comment]
@asranja - pls review the PR

**Review by asranja — APPROVED** [review]

**Pull request merged**